Comp2023/src/main/java/frc/robot/commands/DriveBalance.java

```diff
@@ -0,0 +1,62 @@
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj.DataLogManager;
+import edu.wpi.first.wpilibj.RobotState;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.Swerve;
+
+/**
+ *
+ */
+public class DriveBalance extends CommandBase
+{
+  private Swerve m_swerve;
+  private Timer  m_timer = new Timer( );
+
+  public DriveBalance(Swerve swerve)
+  {
+    m_swerve = swerve;
+
+    addRequirements(m_swerve);
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize( )
+  {
+    DataLogManager.log("Running Drive Balance!");
+    m_timer.reset( );

```

Comment:
I just looked at the Timer class and they've added a new method that makes this quicker and more obvious. The restart() method does both a reset() and a start().

Reply:
The API Docs says that the timer is restarted by stopping it and if not stopped, resetting it and then starting the timer again. Would that mean that if the timer is already stopped it wouldn't start again? Would we keep reset / start in that case?

Reply:
If you click on the method names in the Java API docs, you can see the actual source code for restart().

`
 /**
  * Restart the timer by stopping the timer, if it is not already stopped, resetting the
  * accumulated time, then starting the timer again. If you want an event to periodically reoccur
  * at some time interval from the start time, consider using advanceIfElapsed() instead.
  */
 public void restart() {
   if (m_running) {
     stop();
   }
   reset();
   start();
 }
`